rewriter/Prop.cc

```diff
@@ -504,7 +506,15 @@ vector<ast::ExpressionPtr> processProp(core::MutableContext ctx, PropInfo &ret,
         auto arg = ast::MK::KeywordArgWithDefault(nameLoc, core::Names::allowDirectMutation(), ast::MK::Nil(loc));
         ast::MethodDef::Flags fkFlags;
         fkFlags.discardDef = true;
-        auto fkMethodDef = ast::MK::SyntheticMethod1(loc, loc, fkMethod, std::move(arg),
+
+        core::LocOffsets methodLoc;
+        if (ret.foreignKwLit != nullptr) {
```

Comment:
How is it possible to have `ret.foreign` not be `nullptr`, but to have `ret.foreignKwLit` be `nullptr`?

Reply:
Because of `merchant_token_prop`, which creates an implicit foreign prop